app/web/features/profile/view/UserTextAndLabel.tsx

```diff
@@ -41,6 +46,60 @@ export const LabelsReferencesLastActive = ({ user }: Props) => (
   </>
 );
 
+export const LabelsResponseRate = ({ user }: Props) => {
+  const { t } = useTranslation("profile");
+  const query = useQuery(responseRateKey(user.userId), () =>
+    service.requests.getResponseRate(user.userId)
+  );
+  const rateText =
+    query.data?.responseRate === ResponseRate.RESPONSE_RATE_INSUFFICIENT_DATA
+      ? t("response_rate_text_insufficient")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_LOW
+      ? t("response_rate_text_low")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_SOME
+      ? t("response_rate_text_some")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_MOST
+      ? t("response_rate_text_most")
+      : query.data?.responseRate === ResponseRate.RESPONSE_RATE_ALMOST_ALL
+      ? t("response_rate_text_almost_all")
+      : undefined;
+  const timeText =
+    query.data?.responseRate === ResponseRate.RESPONSE_RATE_SOME
+      ? t("response_time_text_some", {
+          p33: dayjs
+            .duration(query.data.responseTimeP33!.seconds, "second")
```

Comment:
According to the comments in the proto:
https://github.com/Couchers-org/couchers/blob/23b406054919a57b7d39eae12d8ecf943edbf863/app/proto/requests.proto#L165-L172

Shouldn't we be checking for `responseTimeP33` and `responseTimeP66` rather than assuming it's always non-null with `!` if it's sometimes not set by the backend?

Reply:
As is still follows the comments, as we assume it's not null in the cases it says it will be set. I agree it's not very nice though, I'll have a think

Reply:
Yeah, I thought about this too. It's not in general good to create APIs that can represent impossible states.

A better option would be to have response rate be a `oneof` where each of the different choices is its own message so that if response rate is `some`, p66 isn't even there.

I can change it if you want?

Reply:
Sounds good

Reply:
You can do it on this branch if you want

Reply:
OK, I'll do it in a bit

Reply:
@lucaslcode: done

Reply:
Ahh that's fair, overlooked that you were only checking if the response rate is "some" or higher, but the new API response is probably clearer to understand without the impossible states 👍